migration/migrator/migrations/course/20220316135223_add_users_registration_type_column.py

```diff
@@ -0,0 +1,36 @@
+"""Migration for a given Submitty course database."""
+
+
+def up(config, database, semester, course):
+    """
+    Run up migration.
+
+    :param config: Object holding configuration details about Submitty
+    :type config: migrator.config.Config
+    :param database: Object for interacting with given database for environment
+    :type database: migrator.db.Database
+    :param semester: Semester of the course being migrated
+    :type semester: str
+    :param course: Code of course being migrated
+    :type course: str
+    """
+    database.execute("ALTER TABLE users ADD COLUMN IF NOT EXISTS registration_type VARCHAR(255) DEFAULT 'graded';")
+    database.execute("UPDATE users SET registration_type = 'staff' WHERE user_group < 4 AND user_group >= 1")
+    database.execute("ALTER TABLE users DROP CONSTRAINT IF EXISTS check_registration_type;")
+    database.execute("ALTER TABLE users ADD CONSTRAINT check_registration_type CHECK (registration_type in ('graded','audit','withdrawn','staff'));")
+
+
+def down(config, database, semester, course):
+    """
+    Run down migration (rollback).
+
+    :param config: Object holding configuration details about Submitty
+    :type config: migrator.config.Config
+    :param database: Object for interacting with given database for environment
+    :type database: migrator.db.Database
+    :param semester: Semester of the course being migrated
+    :type semester: str
+    :param course: Code of course being migrated
+    :type course: str
+    """
+    pass
```

Comment:
may want to drop the row and revert the constraint changes here

Reply:
I was thinking about that too but it says in the migration documentation 

> In general, your down migration should only undo the necessary changes to make earlier versions of the Submitty software work. For example, if your migration is adding a column to an existing table in the database, it is probably not necessary or desirable to delete that data in the down function. We don’t want to lose any data since we likely plan to return to this state in the future. Your down function may be empty.

`

And then my `up` function drops the constraint for in case anyone changes it